src/test/java/codeu/controller/ControlPanelServletTest.java

```diff
@@ -0,0 +1,172 @@
+    userList.add(notAdmin);
+    userList.add(admin);
+    userList.add(wordy);
+
+    mockRequest = Mockito.mock(HttpServletRequest.class);
+    mockResponse = Mockito.mock(HttpServletResponse.class);
+
+    mockSession = Mockito.mock(HttpSession.class);
+    // When AdminServlet tries to get the session, just return the mock session.
+    Mockito.when(mockRequest.getSession()).thenReturn(mockSession);
+
+    mockRequestDispatcher = Mockito.mock(RequestDispatcher.class);
+    /* When AdminServlet tries to forward to the dispatcher of "/WEB-INF/view/admin.jsp", just
+     * return the mock dispatcher.
+     */
+    Mockito.when(mockRequest.getRequestDispatcher("/WEB-INF/view/control_panel.jsp"))
+        .thenReturn(mockRequestDispatcher);
+
+    mockUserStore = Mockito.mock(UserStore.class);
+    Mockito.when(mockUserStore.getUser("notadmin")).thenReturn(notAdmin);
+    Mockito.when(mockUserStore.getUser("admin")).thenReturn(admin);
+    Mockito.when(mockUserStore.getUser("wordy")).thenReturn(wordy);
+    Mockito.when(mockUserStore.getUser("nobody")).thenReturn(null);
+    Mockito.when(mockUserStore.getUserList()).thenReturn(userList);
+    Mockito.when(mockUserStore.Count()).thenReturn(3);
+
+    controlPanelServlet.setUserStore(mockUserStore);
+
+  }
+
+  @Test
+  public void testDoGet_NoUser() throws IOException, ServletException {
+    Mockito.when(mockSession.getAttribute("user")).thenReturn("nobody");
+
+    controlPanelServlet.doGet(mockRequest, mockResponse);
+
+    // Verify user is *NOT* forwarded to control panel page
+    Mockito.verify(mockRequestDispatcher, Mockito.never()).forward(mockRequest, mockResponse);
+
+    // Verify user is redirected to login page
+    Mockito.verify(mockResponse).sendRedirect("/login");
+  }
+
+  @Test
+  public void testDoGet_NotAdminUsername() throws IOException, ServletException {
+    Mockito.when(mockSession.getAttribute("user")).thenReturn("notadmin");
+
+    controlPanelServlet.doGet(mockRequest, mockResponse);
+
+    // Verify user is *NOT* forwarded to control panel page
+    Mockito.verify(mockRequestDispatcher, Mockito.never()).forward(mockRequest, mockResponse);
+
+    // Verify user is redirected to index page
+    Mockito.verify(mockResponse).sendRedirect("/");
+  }
+
+  @Test
+  public void testDoGet_AdminUsername() throws IOException, ServletException {
+
+    Mockito.when(mockSession.getAttribute("user")).thenReturn("admin");
+
+    controlPanelServlet.doGet(mockRequest, mockResponse);
+
+    Mockito.verify(mockRequest).setAttribute("userList", userList);
+
+    Mockito.verify(mockRequestDispatcher).forward(mockRequest, mockResponse);
+  }
+
+  @Test
+  public void testDoPost_NoUser() throws IOException, ServletException {
+    Mockito.when(mockSession.getAttribute("user")).thenReturn("nobody");
+
+    controlPanelServlet.doPost(mockRequest, mockResponse);
+
+
+  }
+
+  @Test
+  public void testDoPost_NotAdminUsername() throws IOException, ServletException {
+    ;
```

Comment:
Add TODO or code to these two test cases

Reply:
Done!